sdk/messaging/azservicebus/internal/exported/error.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package exported
+
+import "fmt"
+
+// Code is an error code, usable by consuming code to work with
+// programatically.
+type Code string
+
+const (
+	// CodeConnectionLost means our connection was lost and all retry attempts failed.
+	// This typically reflects an extended outage or connection disruption and may
+	// require manual intervention.
+	CodeConnectionLost Code = "connlost"
+
+	// CodeLockLost means that the lock token you have for a message has expired.
+	// This message will be available again after the lock period expires, or, potentially
+	// go to the dead letter queue if delivery attempts have been exceeded.
+	CodeLockLost Code = "locklost"
+)
+
+// Error represents a Service Bus specific error.
+// NOTE: the Code is considered part of the published API but the message that
+// comes back from Error(), as well as the underlying wrapped error, are NOT and
+// are subject to change.
+type Error struct {
+	// Code is a stable error code which can be used as part of programatic error handling.
+	// The codes can expand in the future, but the values (and their meaning) will remain the same.
+	Code     Code
+	innerErr error
+}
+
+// Error is an error message containing the code and a user friendly message, if any.
+func (e Error) Error() string {
```

Comment:
`(e *Error)` I'm surprised the linter didn't flag this.

Reply:
I am too, actually, bit it's still valid I suppose. I switched it over and added in a couple of related tests.